core/services/network_checker.rb

```diff
@@ -45,26 +48,18 @@ def self.load_resources
     YAML.safe_load(File.read(path))
   end
 
-  def self.check_resource?(tool, machine_configurator, machine, resource, logger)
+  def self.check_resource?(tool, machine_configurator, machine, resource, logger, availability_table)
     result = case tool
              when :curl
                check_resource_by_curl?(machine_configurator, machine, resource, logger)
              else
                check_resource_by_wget?(machine_configurator, machine, resource, logger)
              end
-    if result
-      logger.debug("#{resource} is available on the remote server")
-    else
-      logger.error("#{resource} is not available on the remote server")
-    end
-    result
-  end
-
-  # Randomly selects three resources
-  # @return [Array<String>] the list of resources
-  def self.random_resources(resources)
-    shuffle_resources = resources.shuffle
-    shuffle_resources.first(3)
+    availability_table[resource] = if result
+                                     'Ok'
```

Comment:
Do not use strings to identify internal states. You must internally use booleans and convert to strings only when printing the table.